tests/periph_i2c/main.c

```diff
@@ -20,270 +20,547 @@
+            i2c_dev, reg);
+        printf("0x%02x", data);
+        puts("]");
+        return 0;
+    }
+    return print_i2c_error(res);
 }
 
-int cmd_write(int argc, char **argv)
-{
+int cmd_i2c_read_regs(int argc, char **argv) {
     int res;
     uint8_t addr;
-    int length = argc - 2;
+    uint8_t reg;
+    uint8_t flags = 0;
+    int len;
     uint8_t data[BUFSIZE];
 
-    if (i2c_dev < 0) {
-        puts("Error: no I2C device was initialized");
+    if (check_dev() == 1){
         return 1;
     }
-    if (argc < 3) {
-        puts("Error: not enough arguments given");
-        printf("Usage:\n%s: ADDR BYTE0 [BYTE1 [BYTE_n [...]]]\n", argv[0]);
+
+    if (argc < 4 || argc > 5) {
+        INVALID_ARGS;
+        printf("Usage:\n%s ADDR REG LENGTH [FLAG]\n", argv[0]);
         return 1;
     }
 
     addr = atoi(argv[1]);
-    for (int i = 0; i < length; i++) {
-        data[i] = atoi(argv[i + 2]);
+    reg = atoi(argv[2]);
+    len = atoi(argv[3]);
+    if (argc == 5) {
+        flags = atoi(argv[4]);
     }
 
-    if (length == 1) {
-        printf("i2c_write_byte(I2C_%i, 0x%02x, 0x%02x)\n", i2c_dev, addr, data[0]);
-        res = i2c_write_byte(i2c_dev, addr, data[0]);
+    if (len < 1 || len > (int)BUFSIZE) {
+        puts("Error: invalid LENGTH parameter given");
+        return 1;
     }
     else {
-        printf("i2c_write_bytes(I2C_%i, 0x%02x, [", i2c_dev, addr);
-        for (int i = 0; i < length; i++) {
-            printf(", 0x%02x", data[i]);
+        printf("Command: i2c_read_regs(i2c_%i, 0x%02x, 0x%02x, %i)\n", i2c_dev,
+            addr, reg, len);
+        res = i2c_read_regs(i2c_dev, addr, reg, data, len, flags);
+    }
+
+    if (res == I2C_ACK) {
+        printf("Success: i2c_%i successfully read %i bytes from reg 0x%02x: [",
+            i2c_dev, len, reg);
+        for (int i = 0; i < len; i++) {
+            if (i != 0){
+                printf(", ");
+            }
+            printf("0x%02x", data[i]);
         }
-        puts("])");
-        res = i2c_write_bytes(i2c_dev, addr, data, length);
+        puts("]");
+        return 0;
     }
+    return print_i2c_error(res);
+}
+
+int cmd_i2c_read_byte(int argc, char **argv) {
+    int res;
+    uint16_t addr;
+    uint8_t flags = 0;
+    uint8_t data;
 
-    if (res < 0) {
-        puts("Error: no bytes were written");
+    if (check_dev() == 1){
         return 1;
     }
-    else {
-        printf("I2C_%i: successfully wrote %i bytes to the bus\n", i2c_dev, res);
+
+    if (argc < 2 || argc > 3) {
+        INVALID_ARGS;
+        printf("Usage:\n%s ADDR [FLAG]\n", argv[0]);
+        return 1;
+    }
+
+    addr = atoi(argv[1]);
+    if (argc == 3) {
+        flags = atoi(argv[2]);
+    }
+
+    printf("Command: i2c_read_byte(i2c_%i, 0x%02x, 0x%02x)\n", i2c_dev, addr, flags);
+    res = i2c_read_byte(i2c_dev, addr, &data, flags);
+
+    if (res == I2C_ACK) {
+        printf("Success: i2c_%i successfully read 1 byte: [", i2c_dev);
```

Comment:
nit picky: omit `successfully` because its already a `Success` 😉 

applies to other output above and below, too.

Reply:
Yup legacy, will do. [TODO] x3